sync-core/src/main/java/com/cloudant/sync/datastore/PreparedAttachment.java

```diff
@@ -32,6 +32,48 @@
  */
 public class PreparedAttachment {
 
+    // TODO this ctor when the attachment comes from the server
+
+    /**
+     * Prepare an attachment by copying it to a temp location and calculating its sha1.
+     *
+     * @param attachment The attachment to prepare
+     * @param attachmentsDir The 'BLOB store' or location where attachments are stored for this database
+     * @throws AttachmentNotSavedException
+     */
+    public PreparedAttachment(Attachment attachment,
+                              long length,
+                              long encodedLength,
+                              String attachmentsDir) throws AttachmentException {
+        this.attachment = attachment;
+        this.tempFile = new File(attachmentsDir, "temp" + UUID.randomUUID());
+        this.length = length;
+        this.encodedLength = encodedLength;
+        FileInputStream tempFileIS = null;
+        try {
+            FileUtils.copyInputStreamToFile(attachment.getInputStream(), tempFile);
+            long expectedLength;
+            if (this.attachment.encoding == Attachment.Encoding.Plain) {
+                expectedLength = length;
+            } else {
+                expectedLength = encodedLength;
+            }
```

Comment:
Shouldn't the replicator code be making this check? I'm not sure PreparedAttachment needs to check this, it's not its job to validate stuff.